src/praisonai/praisonai/scheduler/shared.py

```
@@ -0,0 +1,60 @@
+def backoff_delay(attempt: int, *, base: float = 2.0, cap: float = 60.0) -> float:
+    """Exponential backoff, capped. Used by both sync & async schedulers."""
+    return min(base ** attempt, cap)
+
+
+def safe_call(cb, *args) -> None:
+    """Run a user callback without letting it tear the scheduler down."""
+    if cb is None:
+        return
+    try:
+        cb(*args)
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)